src/abstract-field/abstract-field.component.ts

```diff
@@ -62,6 +63,19 @@ export abstract class AbstractFieldComponent
     this.externalCategorizedErrorSubscription.unsubscribe();
   }
 
+  /**
+   * Gets path for child, returns from `pathCache` if it is a hit
+   * in order not to re-render child component due to reference change its path.
+   *
+   * @param key - index or field name for child
+   */
+  getPathForChild(key: any): Array<any> {
+    if (!this.pathCache[key] || this.pathCache[key][this.path.length - 1] !== this.path[this.path.length - 1]) {
```

Comment:
`this.pathCache[key][this.path.length - 1] !== this.path[this.path.length - 1]` part is the trick that fixes the issue here, others are refactoring.